rtnetlink/src/link/add.rs

```diff
@@ -17,6 +17,137 @@ use crate::{
     Handle,
 };
 
+/// A request to create a new vxlan link. This is equivalent to the `ip link add NAME vxlan id ID ...` commands.
+///
+/// It provides methods to customize the creation of
+/// the vxlan interface
+pub struct VxlanAddRequest {
+    request: LinkAddRequest,
+    info_data: Vec<InfoVxlan>,
+}
+
+impl VxlanAddRequest {
+    pub async fn execute(self) -> Result<(), Error> {
+        let s = self
+            .request
+            .link_info(InfoKind::Vxlan, Some(InfoData::Vxlan(self.info_data)));
+        s.execute().await
+    }
+
+    pub fn up(mut self) -> Self {
+        self.request = self.request.up();
+        self
+    }
+
+    pub fn link(mut self, index: u32) -> Self {
+        self.info_data.push(InfoVxlan::Link(index));
+        self
+    }
+
+    pub fn port(mut self, port: u16) -> Self {
+        self.info_data.push(InfoVxlan::Port(port));
+        self
+    }
+
+    pub fn group(mut self, addr: std::net::Ipv4Addr) -> Self {
+        let octets = addr.octets();
+        let ip: u32 = (octets[0] as u32) << 24
+            & (octets[1] as u32) << 16
+            & (octets[2] as u32) << 8
+            & (octets[3] as u32);
+        self.info_data.push(InfoVxlan::Group(ip));
+        self
+    }
+
+    pub fn group6(mut self, addr: std::net::Ipv6Addr) -> Self {
+        let ip: u128 = u128::from_le_bytes(addr.octets());
+        self.info_data.push(InfoVxlan::Group6(ip));
+        self
+    }
+
+    pub fn local(mut self, addr: std::net::Ipv4Addr) -> Self {
+        let octets = addr.octets();
+        let ip: u32 = (octets[0] as u32) << 24
```

Comment:
Maybe:
```rust
u32::from_bytes_be(addr.octets())
```
(if I got my endianness right)

Reply:
I was thinking that maybe instead of using having hard-coded BE or LE, I would like to check the machine endianness and use the correct one.
I was looking at this: https://docs.rs/byteordered/0.5.0/byteordered/enum.Endianness.html 

If you are ok with this I can use that crate as a dependency and use always the correct endianness